website/blog/2021-05-25-inking.md

```diff
@@ -0,0 +1,412 @@
+
+```xml
+<react:ReactRootView 
+    x:Name="reactRootView"
+    ComponentName="InkCanvasv064"
+    IsPerspectiveEnabled="False"
+    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
+    MinHeight="400"/>
+```
+
+That's it. Now we can continue the implementation of the features of our custom control.
+
+### Exposing a property
+
+This isn't stricly related to enable inking, but it's a good learning when it comes to build native UI modules. The goal of this section is to expose the dependency property called `Title` that we have created to JavaScript, so that we can customize it using JSX. Properties can be exposed using the `ViewManagerProperty` attribute, like in the following sample:
+
+```csharp
+[ViewManagerProperty("title")]
+public void SetTitle(InkCanvasControl view, string value)
+{
+    if (null != value)
+    {
+        view.Title = value;
+    }
+    else
+    {
+        view.ClearValue(InkCanvasControl.TitleProperty);
+    }
+}
+```
+
+The method, called `SetTitle()`, is decorated with the `ViewManagerProperty` attribute, which sets the name that will be used to expose the property to JSX (in this case, it's `title` in lowercase, to follow the JavaScript naming conventions).
+The method receives, as input, a reference to the user control and the value assigned via JSX. The implementation of the method is straigthforward: if the property assigned in JSX has a value, we assign it to the `Title` property of the user control; otherwise, we clear it.
+
+Now we'll be able to leverage the `title` property in our React Native components, like in the following sample:
+
+```javascript
+class App extends React.Component {
+
+  render(){
+    return (
+      <View>
+        <InkCanvas style={{ width: 1000, height: 1000}}
+                   title="Hello world!" />
+      </View>
+    );
+  }
+};
+```
+
+And this is the final result:
+
+![](assets/2021-05-25-inking/HelloWorld.png)
+
+### Exposing a command
+
+One of the most common features that developers want to implement when they work with inking is the ability to save the content of the `InkCanvas` control as an image. To enable this feature we're going to implement a command. The concept is very similar to exposing a method in a React Native module and it will enable us to trigger, from JavaScript, the native APIs exposed by the `InkCanvas` control to render the ink into a stream.
+
+The first step is to install a library called [Win2D](https://github.com/microsoft/Win2D), which enables advanced 2D operations in a UWP app. Thanks to this library, we'll be able to render the content of the canvas into a stream that we can save into a file. To install it, right click on your Windows Runtime Component, choose **Manage NuGet Packages** and look for the [Win2D package](https://www.nuget.org/packages/Win2D.uwp/).
```

Comment:
curious, is this necessary at all? seems like one should be able to save the content of the ink canvas without using win2D (which is also C# only afaik)

Reply:
Let me investigate and see if there's another way. Last time I checked, not using Win2D means basically reimplementing what Win2D does to render the canvas. Unfortunately the control doesn't expose any straightforward API to save the content into an image.